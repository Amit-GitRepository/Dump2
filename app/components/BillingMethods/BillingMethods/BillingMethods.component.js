/* BillingMethods: show all the available paper bill & ebill items*/

import EBillItem from '../EBillItem/EBillItem.component';
import map from 'lodash/map';
import noop from 'lodash/noop';
import PaperBillItem from '../PaperBillItem/PaperBillItem.component';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import result from 'lodash/result';
import styles from './BillingMethods.style';
import {accountTypeMap} from '../../../config/ebill.config';
import {Button, Card} from '../../Shared';
import {ScrollView, View} from 'react-native';
import {translate} from '../../../language/i18n/helper';

class BillingMethods extends Component {
  onEditEBill = (data) => () => {
    this.props.onEditEbillAction({account: data});
  }

  onApplyEbill = () => {
    this.props.onApplyEbillAction();
  }

  generatePaperBillSection = (paperBillContent) => {
    const paperBillItem = map(paperBillContent, (data, index) => (<PaperBillItem
      key={`Paper-Bill-Item-${index}`}
      accountType={accountTypeMap[data.accountType]}
      productId={data.productId}
      billingAddress={result(data, 'billingValue.addressLine1', '') +  result(data, 'billingValue.addressLine2', '') + result(data, 'billingValue.addressLine3', '') + result(data, 'billingValue.addressLine4', '')}
      noBorder={index === paperBillContent.length - 1} />));
    return (
      <Card header={translate('BILLING_METHODS__PAPER_HEADER')}>
        {paperBillItem}
      </Card>
    );
  };

  generateEBillSection = (ebillContent) => {
    const eBillItem = map(ebillContent, (data, index) => (<EBillItem
      key={`E-Bill-Item-${index}`}
      accountType={accountTypeMap[data.accountType]}
      productId={data.productId}
      billingFormat={data.billingFormat}
      ebillValue={data.billingValue}
      onEditBill={this.onEditEBill(data)}
      noBorder={index === ebillContent.length - 1} />));
    return (
      <Card containerStyles={styles.ebillCard} header={translate('BILLING_METHODS__EBILL_HEADER')}>
        {eBillItem}
      </Card>
    );
  }

  generateApplyEbillButton = () => {
    const {paperBillContent} = this.props;
    if (paperBillContent.length > 0) {
      return (<View style={styles.buttonContainer}>
        <Button text={translate('BILLING_METHODS__ADD_APPLY')} onPress={this.onApplyEbill}/>
      </View>);
    }
  }

  render () {
    const {ebillContent, paperBillContent} = this.props;
    return (
      <View style={styles.container}>
        <ScrollView showsVerticalScrollIndicator={false} style={styles.cardContainer} contentContainerStyle={styles.scrollContainer}>
          {ebillContent.length > 0 && this.generateEBillSection(ebillContent)}
          {paperBillContent.length > 0 && this.generatePaperBillSection(paperBillContent)}
        </ScrollView>
        {this.generateApplyEbillButton()}
      </View>
    );
  }
}
BillingMethods.defaultProps = {
  onApplyEbillAction: noop,
  onEditBillAction: noop
};
BillingMethods.propTypes = {
  onApplyEbillAction: PropTypes.func,
  onEditEbillAction: PropTypes.func,
  ebillContent: PropTypes.array,
  paperBillContent: PropTypes.array
};
export default BillingMethods;
