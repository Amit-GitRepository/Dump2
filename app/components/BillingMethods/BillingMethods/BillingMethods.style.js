export default {
  container: {
    padding: 5,
    flex: 1,
    justifyContent: 'space-between'
  },
  cardContainer: {
    marginBottom: 10,
    paddingHorizontal: 12,
    paddingTop: 5
  },
  scrollContainer: {
    paddingBottom: 20
  },
  ebillCard: {
    marginBottom: 20
  },
  buttonContainer: {
    paddingHorizontal: 10
  }
};
