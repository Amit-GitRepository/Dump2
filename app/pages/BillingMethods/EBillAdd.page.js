import EBillAdd from '../../components/BillingMethods/EBillAdd/EBillAdd.component';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import result from 'lodash/result';
import {billFormats} from '../../config/ebill.config';
import {connect} from 'react-redux';
import * as actions from '../../redux/actions/index.actions';

class EBillAddPage extends Component {
  render () {
    const {billPreference, showPopup} = this.props;
    return (
      <EBillAdd
        paperBillContent={billPreference[billFormats.PAPER] || []}
        showPopup={showPopup}
      />
    );
  }
}

EBillAddPage.propTypes = {
  billPreference: PropTypes.object,
  showPopup: PropTypes.func
};

const mapStateToProps = (state) => ({
  billPreference: result(state, 'ebill.billPreference', {})
});

export const mapDispatchToProps = (dispatch) => ({
  showPopup: (value) => dispatch(actions.showPopup(value))
});

export default connect(mapStateToProps, mapDispatchToProps)(EBillAddPage);
