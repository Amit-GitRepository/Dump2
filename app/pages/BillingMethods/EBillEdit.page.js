import assign from 'lodash/assign';
import EBillEdit from '../../components/BillingMethods/EBillEdit/EBillEdit.component';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import result from 'lodash/result';
import {billFormats} from '../../config/ebill.config';
import {connect} from 'react-redux';
import {translate} from '../../language/i18n/helper';
import * as actions from '../../redux/actions/index.actions';

class EBillEditPage extends Component {
  state = {
    account: result(this.props.navigation, 'state.params.account', {})
  }

  saveEbillAccount = (data) => {
    const account = assign({}, data, {
      accountId: this.state.account.accountId,
      accountType: this.state.account.accountType
    });

    let header = translate('BILLING_METHODS__ADD_GO_PAPERLESS');
    let body = translate('BILLING_METHODS__ADD_BODY_PAPERLESS');

    switch (account.billingFormat) {
    case billFormats.PAPER: {
      header = translate('BILLING_METHODS__ADD_GO_PAPER');
      body = translate('BILLING_METHODS__ADD_BODY_PAPER');
      break;
    }
    case billFormats.SMS: {
      body = translate('BILLING_METHODS__ADD_BODY_SMS');
      break;
    }
    case billFormats.EMAIL: {
      body = translate('BILLING_METHODS__ADD_BODY_EMAIL');
      break;
    }
    default: {
      // do nothing
    }
    }

    const message = {
      title: header,
      body: body,
      buttons: [
        {
          title: translate('BILLING_METHODS__ADD_CANCEL')
        },
        {
          title: translate('BILLING_METHODS__ADD_CONFIRM'),
          actionType: actions.APPLY_FOR_EBILL,
          nextAction: [account]
        }
      ]
    };
    this.props.showPopup(message);
  }
  render () {
    const {accountType, billingFormat, productId, billingValue} = this.state.account;
    return (
      <EBillEdit
        accountType={accountType}
        billingFormat={billingFormat}
        productId={productId || ''}
        defaultMSISDN={billingFormat === 'SMS' ? billingValue : (accountType === 'tmhPostpaid' ? productId : null)}
        defaultEmail={billingFormat === 'EMAIL' ? billingValue : null}
        onSaveAction={this.saveEbillAccount}
      />
    );
  }
}

EBillEditPage.propTypes = {
  navigation: PropTypes.object.isRequired,
  showPopup: PropTypes.func
};

const mapStateToProps = () => ({});

export const mapDispatchToProps = (dispatch) => ({
  showPopup: (value) => dispatch(actions.showPopup(value))
});

export default connect(mapStateToProps, mapDispatchToProps)(EBillEditPage);
