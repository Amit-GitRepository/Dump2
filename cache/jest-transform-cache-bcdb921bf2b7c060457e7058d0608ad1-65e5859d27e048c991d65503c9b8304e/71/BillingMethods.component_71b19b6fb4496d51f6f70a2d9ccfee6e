Object.defineProperty(exports,"__esModule",{value:true});var cov_16oygadfeb=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillingMethods/BillingMethods/BillingMethods.component.js',hash='d42eb28804d232dd6eae9d3f466db4f27f9c03d8',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillingMethods/BillingMethods/BillingMethods.component.js',statementMap:{'0':{start:{line:15,column:4},end:{line:27,column:11}},'1':{start:{line:31,column:32},end:{line:31,column:42}},'2':{start:{line:33,column:4},end:{line:47,column:6}},'3':{start:{line:51,column:33},end:{line:51,column:43}},'4':{start:{line:53,column:4},end:{line:66,column:15}},'5':{start:{line:69,column:0},end:{line:72,column:2}},'6':{start:{line:73,column:0},end:{line:76,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:14,column:29},end:{line:14,column:30}},loc:{start:{line:15,column:4},end:{line:27,column:11}},line:15},'1':{name:'(anonymous_1)',decl:{start:{line:30,column:25},end:{line:30,column:26}},loc:{start:{line:30,column:31},end:{line:48,column:3}},line:30},'2':{name:'(anonymous_2)',decl:{start:{line:50,column:2},end:{line:50,column:3}},loc:{start:{line:50,column:12},end:{line:67,column:3}},line:50}},branchMap:{},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0},f:{'0':0,'1':0,'2':0},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();

var _EBillItem=require('../EBillItem/EBillItem.component');var _EBillItem2=_interopRequireDefault(_EBillItem);
var _noop=require('lodash/noop');var _noop2=_interopRequireDefault(_noop);
var _PaperBillItem=require('../PaperBillItem/PaperBillItem.component');var _PaperBillItem2=_interopRequireDefault(_PaperBillItem);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _BillingMethods=require('./BillingMethods.style');var _BillingMethods2=_interopRequireDefault(_BillingMethods);
var _Shared=require('../../Shared');
var _reactNative=require('react-native');
var _helper=require('../../../language/i18n/helper');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

BillingMethods=function(_Component){_inherits(BillingMethods,_Component);function BillingMethods(){var _ref;var _temp,_this,_ret;_classCallCheck(this,BillingMethods);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=BillingMethods.__proto__||Object.getPrototypeOf(BillingMethods)).call.apply(_ref,[this].concat(args))),_this),_this.
generatePaperBillSection=function()
{cov_16oygadfeb.f[0]++;cov_16oygadfeb.s[0]++;return _react2.default.createElement(_Shared.Card,{header:(0,_helper.translate)('BILLING_METHODS__PAPER_HEADER')},
_react2.default.createElement(_PaperBillItem2.default,{
accountType:'TrueMove H Postpaid',
serviceID:'9876543210',
billingAddress:'444 Soi Ladprao 109, Ladprao rd., Bangkapi, Bangkok 10240'}),

_react2.default.createElement(_PaperBillItem2.default,{
accountType:'True Online',
serviceID:'9876543210',
billingAddress:'444 Soi Ladprao 109, Ladprao rd., Bangkapi, Bangkok 10240',
noBorder:true}));

},_this.


generateEBillSection=function(){cov_16oygadfeb.f[1]++;var _ref2=(cov_16oygadfeb.s[1]++,
_this.props),onEditEbillAction=_ref2.onEditEbillAction;cov_16oygadfeb.s[2]++;

return(
_react2.default.createElement(_Shared.Card,{
containerStyles:_BillingMethods2.default.ebillCard,
header:(0,_helper.translate)('BILLING_METHODS__EBILL_HEADER')},

_react2.default.createElement(_EBillItem2.default,{
accountType:'True Vision',
serviceID:'9876543210',
ebillFormat:'SMS',
ebillValue:'9876543210',
onEditBill:onEditEbillAction,
noBorder:true})));



},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(BillingMethods,[{key:'render',value:function render()

{cov_16oygadfeb.f[2]++;var _ref3=(cov_16oygadfeb.s[3]++,
this.props),onApplyEbillAction=_ref3.onApplyEbillAction;cov_16oygadfeb.s[4]++;

return(
_react2.default.createElement(_reactNative.View,{style:_BillingMethods2.default.container},
_react2.default.createElement(_reactNative.ScrollView,{
showsVerticalScrollIndicator:false,
style:_BillingMethods2.default.cardContainer,
contentContainerStyle:_BillingMethods2.default.scrollContainer},

this.generateEBillSection(),
this.generatePaperBillSection()),

_react2.default.createElement(_reactNative.View,{style:_BillingMethods2.default.buttonContainer},
_react2.default.createElement(_Shared.Button,{text:(0,_helper.translate)('BILLING_METHODS__ADD_APPLY'),onPress:onApplyEbillAction}))));


}}]);return BillingMethods;}(_react.Component);cov_16oygadfeb.s[5]++;

BillingMethods.defaultProps={
onApplyEbillAction:_noop2.default,
onEditBillAction:_noop2.default};cov_16oygadfeb.s[6]++;

BillingMethods.propTypes={
onApplyEbillAction:_propTypes2.default.func,
onEditEbillAction:_propTypes2.default.func};exports.default=

BillingMethods;