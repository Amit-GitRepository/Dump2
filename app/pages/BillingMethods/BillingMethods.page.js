import BillingMethods from '../../components/BillingMethods/BillingMethods/BillingMethods.component';
import ComponentWithTab from '../../components/ComponentWithTab/ComponentWithTab.component';
import PropTypes from 'prop-types';
import React from 'react';
import {billFormats} from '../../config/ebill.config';
import {connect} from 'react-redux';
import {every, forEach, map, result, union, uniq} from 'lodash';
import {NavigationActions} from 'react-navigation';
import {View} from 'react-native';
import * as actions from '../../redux/actions/index.actions';

class BillingMethodsPage extends ComponentWithTab {

  componentWillMount () {
    this.props.getBillingPreference(this.getAccountIds(this.props.productList));
  }

  getAccountIds = (productList) => {
    const accountIds = [];
    forEach(productList, (array, key) => {
      if (key === 'tmhPrepaid') {
        return;
      }

      if (key === 'conv') {
        // so for convergence account - All the accounts for which
        // convergenceCode is same && isBillConsolidated
        // then use any one of the accountIds
        // otherwise use all the accountIds individually
        map(array, (val) => {
          const productList = result(val, 'products', []);
          if (every(productList, (item) => item.isBillConsolidated)) {
            accountIds.push(productList[0].accountId);
          } else {
            map(productList, (item) => {
              accountIds.push(item.accountId);
            });
          }
        });
      } else {
        map(array, (val) => {
          accountIds.push(val.accountId);
        });
      }
    });
    return uniq(accountIds);
  }

  navigateToScreen = (screen) => (params = {}) => {
    this.props.goToScreen(screen, params);
  }

  renderWithTab () {
    const {billPreference} = this.props;
    return (
      <View>
        <BillingMethods
          ebillContent={union(billPreference[billFormats.EMAIL] || [], billPreference[billFormats.SMS] || [])}
          paperBillContent={billPreference[billFormats.PAPER] || []}
          onApplyEbillAction={this.navigateToScreen('EbillAdd')}
          onEditEbillAction={this.navigateToScreen('EbillEdit')}
        />
      </View>
    );
  }
}

BillingMethodsPage.propTypes = {
  goToScreen: PropTypes.func,
  productList: PropTypes.object,
  billPreference: PropTypes.object,
  getBillingPreference: PropTypes.func
};

const mapStateToProps = (state) => ({
  productList: result(state, 'billUsage.productList', {}),
  billPreference: result(state, 'ebill.billPreference', {})
});

export const mapDispatchToProps = (dispatch) => ({
  goToScreen: (screenName, params) => dispatch(NavigationActions.navigate({routeName: screenName, params})),
  getBillingPreference: (accountIds) => dispatch(actions.getBillingPreference(accountIds))
});

export default connect(mapStateToProps, mapDispatchToProps)(BillingMethodsPage);
