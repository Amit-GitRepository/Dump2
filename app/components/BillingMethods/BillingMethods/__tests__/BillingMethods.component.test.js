import BillingMethods from '../BillingMethods.component';
import React from 'react';
import renderer from 'react-test-renderer';

describe('EBillAdd component', () => {
  it('renders correctly', () => {
    const billPreference = {
      'PAPER': [
        {
          'accountId': '10100949',
          'billingFormat': 'PAPER',
          'billingValue': {
            'country': '',
            'zipCode': '10170',
            'roomNo': '',
            'city': 'กรุงเทพมหานคร',
            'addressType': 'I',
            'moo': '',
            'soi': 'ซอยพุทธมณฑลสาย 2 ซอย 24',
            'building': '',
            'street': '-',
            'district': 'เขตทวีวัฒนา',
            'addressLine1': '11/9 * * ซอยพุทธมณฑลสาย 2 ซอย 24 *',
            'houseNo': '11/9',
            'addressLine2': '* * * * -',
            'timeAtAddress': '0101',
            'addressLine3': '* แขวงศาลาธรรมสพน์ * เขตทวีวัฒนา',
            'addressLine4': 'กรุงเทพมหานคร 10170',
            'floor': '',
            'subDistrict': 'แขวงศาลาธรรมสพน์'
          },
          'productId': '0968730312',
          'subscriberId': '193510',
          'balance': '0.0',
          'subscriptionType': 'POSTPAID',
          'statusCode': 'ACTIVE',
          'productType': 'TrueMoveH',
          'accountType': 'tmhPostpaid'
        }
      ],
      'SMS': [
        {
          'accountId': '10101029',
          'billingFormat': 'SMS',
          'billingValue': '0855469856',
          'productId': '0968730313',
          'subscriberId': '193591',
          'balance': '0.0',
          'subscriptionType': 'POSTPAID',
          'statusCode': 'ACTIVE',
          'productType': 'TrueMoveH',
          'accountType': 'tmhPostpaid'
        }
      ]
    };
    const component = renderer.create(<BillingMethods ebillContent={billPreference.SMS} paperBillContent={billPreference.PAPER} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
