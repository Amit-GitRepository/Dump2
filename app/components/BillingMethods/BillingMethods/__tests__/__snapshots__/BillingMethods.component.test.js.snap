// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EBillAdd component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-between",
      "padding": 5,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingBottom": 20,
      }
    }
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "marginBottom": 10,
        "paddingHorizontal": 12,
        "paddingTop": 5,
      }
    }
  >
    <View>
      <View
        elevation={5}
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 8,
              "flex": 1,
              "marginHorizontal": -1,
              "overflow": "hidden",
              "shadowColor": "#303C46",
              "shadowOffset": Object {
                "height": 2,
                "width": 1,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 1,
            },
            Object {
              "marginBottom": 20,
            },
          ]
        }
      >
        <View
          onPress={null}
          style={
            Array [
              Object {
                "zIndex": 1,
              },
              Object {},
            ]
          }
        >
          <View>
            <View
              style={
                Object {
                  "borderWidth": 0,
                  "elevation": 3,
                  "shadowColor": "#303C46",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 1,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderBottomColor": "#F93E60",
                      "paddingHorizontal": 5,
                    },
                    Object {
                      "borderBottomColor": "#303C46",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#E9E9EA",
                          "padding": 10,
                        },
                        Object {
                          "backgroundColor": "#303C46",
                          "marginHorizontal": -5,
                          "padding": 10,
                        },
                      ],
                      Object {},
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#303C46",
                        },
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 18,
                          },
                          Object {
                            "color": "#FFFFFF",
                          },
                        ],
                        Object {
                          "fontFamily": "SukhumvitSet-Bold",
                        },
                      ]
                    }
                  >
                    e-Bill
                  </Text>
                </View>
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#303C46",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "bottom": 9.6,
                      "marginBottom": -9.6,
                    },
                    Object {},
                  ],
                  Object {
                    "fontFamily": "iservice",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              Ó§Å
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "padding": 15,
              },
              null,
            ]
          }
        >
          <View
            style={
              Object {
                "marginBottom": 5,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                  "fontFamily": "SukhumvitSet-Bold",
                  "fontSize": 22,
                }
              }
            >
              0968730313
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#FBBC1E",
                  "fontFamily": "SukhumvitSet-Bold",
                }
              }
            >
              True Move H postpaid
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                  "fontFamily": "SukhumvitSet-Bold",
                  "fontSize": 18,
                  "paddingTop": 4,
                }
              }
            >
              SMS
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                  "fontFamily": "SukhumvitSet-Text",
                  "fontSize": 16,
                  "paddingTop": 4,
                }
              }
            >
              0855469856
            </Text>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "transparent",
                  "borderRadius": 50,
                  "borderWidth": 0,
                  "opacity": 1,
                  "padding": 5,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "stretch",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#303C46",
                        },
                        Array [
                          Object {
                            "color": "#F93E60",
                            "fontSize": 16,
                          },
                          Object {},
                        ],
                        Object {
                          "fontFamily": "SukhumvitSet-SemiBold",
                        },
                      ]
                    }
                  >
                    Edit
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        elevation={5}
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 8,
              "flex": 1,
              "marginHorizontal": -1,
              "overflow": "hidden",
              "shadowColor": "#303C46",
              "shadowOffset": Object {
                "height": 2,
                "width": 1,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 1,
            },
            Object {},
          ]
        }
      >
        <View
          onPress={null}
          style={
            Array [
              Object {
                "zIndex": 1,
              },
              Object {},
            ]
          }
        >
          <View>
            <View
              style={
                Object {
                  "borderWidth": 0,
                  "elevation": 3,
                  "shadowColor": "#303C46",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 1,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderBottomColor": "#F93E60",
                      "paddingHorizontal": 5,
                    },
                    Object {
                      "borderBottomColor": "#303C46",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#E9E9EA",
                          "padding": 10,
                        },
                        Object {
                          "backgroundColor": "#303C46",
                          "marginHorizontal": -5,
                          "padding": 10,
                        },
                      ],
                      Object {},
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#303C46",
                        },
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 18,
                          },
                          Object {
                            "color": "#FFFFFF",
                          },
                        ],
                        Object {
                          "fontFamily": "SukhumvitSet-Bold",
                        },
                      ]
                    }
                  >
                    Paper Bill
                  </Text>
                </View>
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#303C46",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "bottom": 9.6,
                      "marginBottom": -9.6,
                    },
                    Object {},
                  ],
                  Object {
                    "fontFamily": "iservice",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              Ó§Å
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "padding": 15,
              },
              null,
            ]
          }
        >
          <View
            style={
              Object {
                "marginBottom": 5,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                  "fontFamily": "SukhumvitSet-Bold",
                  "fontSize": 22,
                }
              }
            >
              0968730312
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#FBBC1E",
                  "fontFamily": "SukhumvitSet-Bold",
                }
              }
            >
              True Move H postpaid
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#303C46",
                    "fontFamily": "SukhumvitSet-Bold",
                    "fontSize": 18,
                    "paddingTop": 4,
                  }
                }
              >
                Billing Address
              </Text>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                  "fontFamily": "SukhumvitSet-Text",
                  "fontSize": 16,
                }
              }
            >
              11/9 * * ‡∏ã‡∏≠‡∏¢‡∏û‡∏∏‡∏ó‡∏ò‡∏°‡∏ì‡∏ë‡∏•‡∏™‡∏≤‡∏¢ 2 ‡∏ã‡∏≠‡∏¢ 24 ** * * * -* ‡πÅ‡∏Ç‡∏ß‡∏á‡∏®‡∏≤‡∏•‡∏≤‡∏ò‡∏£‡∏£‡∏°‡∏™‡∏û‡∏ô‡πå * ‡πÄ‡∏Ç‡∏ï‡∏ó‡∏ß‡∏µ‡∏ß‡∏±‡∏í‡∏ô‡∏≤‡∏Å‡∏£‡∏∏‡∏á‡πÄ‡∏ó‡∏û‡∏°‡∏´‡∏≤‡∏ô‡∏Ñ‡∏£ 10170
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#F93E60",
          "borderRadius": 50,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                },
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 16,
                  },
                  Object {},
                ],
                Object {
                  "fontFamily": "SukhumvitSet-Bold",
                },
              ]
            }
          >
            Apply e-Bill
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
